fhir-model/src/main/java/org/linuxforhealth/fhir/model/string/util/StringSizeControlStrategyFactory.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * (C) Copyright IBM Corp. 2022
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.linuxforhealth.fhir.model.string.util;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.ServiceLoader;
+
+import org.linuxforhealth.fhir.exception.FHIROperationException;
+import org.linuxforhealth.fhir.model.resource.OperationOutcome;
+import org.linuxforhealth.fhir.model.string.util.strategy.StringSizeControlStrategy;
+import org.linuxforhealth.fhir.model.type.code.IssueType;
+import org.linuxforhealth.fhir.model.util.FHIRUtil;
+
+/**
+ * Controls the creation of the StringSizeControlStrategy objects using the ServiceLoader.
+ */
+public class StringSizeControlStrategyFactory {
```

Comment:
seems a bit overcomplicated / overengineered to me, but maybe this will pay off in the future ?

Reply:
As discussed with Robin, this has been changed to a Strategy pattern interface and an implementation class and the instantiation of the implementation class through the service loader has been removed.